modules.go

```diff
@@ -0,0 +1,455 @@
+package goja
+
+import (
+	"errors"
+)
+
+type HostResolveImportedModuleFunc func(referencingScriptOrModule interface{}, specifier string) (ModuleRecord, error)
```

Comment:
Is currently being provided to each call that needs it, but it's both needed during parsing and during evaluation (unless we change something significantly). So setting it on the Runtime will mean we need a Runtime for the parsing that isn't really nice. If we just add it to Runtime for the Evaluation it will not be that big of an improvement. 

But I am for any suggestions.